osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -164,11 +166,15 @@ public abstract class DrawableHitObject : SkinReloadableDrawable
         /// </summary>
         /// <param name="initialHitObject">
         /// The <see cref="HitObject"/> to be initially applied to this <see cref="DrawableHitObject"/>.
-        /// If <c>null</c>, a hitobject is expected to be later applied via <see cref="Apply"/> (or automatically via pooling).
+        /// If <c>null</c>, a hitobject is expected to be later applied via <see cref="Apply(osu.Game.Rulesets.Objects.HitObjectLifetimeEntry)"/> (or automatically via pooling).
         /// </param>
         protected DrawableHitObject([CanBeNull] HitObject initialHitObject = null)
         {
-            HitObject = initialHitObject;
+            if (initialHitObject != null)
+            {
+                lifetimeEntry = new UnmanagedHitObjectEntry(initialHitObject);
+                ensureEntryHasResult();
```

Comment:
Is this necessary here? Won't eventual application of the entry ensure the existence of the result anyways?

Reply:
It is not currently necessary.
I thought having to worry about null Result is another source of a bug.
Ideally `HitObjectLifetimeEntry.Result` should be readonly and set in the constructor, but the current code doesn't allow this, and it is a compromise.